tests/main/interfaces-kernel-module-control/task.yaml

```diff
@@ -0,0 +1,76 @@
+summary: Ensure that the kernel-module-control interface works.
+
+details: |
+    The kernel-module-control interface allows insertion, removal and querying
+    of modules.
+
+    A snap which defines a kernel-module-control plug must be shown in the
+    interfaces list. The plug must not be autoconnected on install and, as
+    usual, must be able to be reconnected.
+
+    A snap declaring a plug on this interface must be able to list the modules
+    loaded, insert and remove a module. For the test we use the floppy module.
+
+prepare: |
+    echo "Given a snap declaring a plug on the kernel-module-control interface is installed"
+    snap install --edge test-snapd-kernel-module-control-consumer
+
+restore: |
+    rm -f list.error remove.error
+    if ! lsmod | MATCH floppy; then
+        insmod /lib/modules/$(uname -r)/kernel/drivers/block/floppy.ko
+    fi
+
+execute: |
+    CONNECTED_PATTERN=":kernel-module-control +test-snapd-kernel-module-control-consumer"
+    DISCONNECTED_PATTERN="\- +test-snapd-kernel-module-control-consumer:kernel-module-control"
+
+    echo "The plug is disconnected by default"
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "==================================="
+
+    echo "When the plug is connected"
+    snap connect test-snapd-kernel-module-control-consumer:kernel-module-control
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then the snap is able to list the existing modules"
+    [ $(su -l -c "test-snapd-kernel-module-control-consumer.lsmod" test | wc -l) -gt 2 ]
+
+    echo "And the snap is able to remove a module"
+    lsmod | MATCH floppy
```

Comment:
Curious, do we have floppy on all the arches that we run autopkgtest on :) - ppc64el ?